components/org.wso2.carbon.identity.data.publisher.authentication.analytics.login/src/main/java/org/wso2/carbon/identity/data/publisher/authentication/analytics/login/AnalyticsLoginDataPublisherUtils.java

```diff
@@ -82,7 +82,11 @@ public static AuthenticationData buildAuthnDataForAuthnStep(Event event) {
         authenticationData.setEventId(UUID.randomUUID().toString());
         authenticationData.setEventType(AnalyticsLoginDataPublishConstants.STEP_EVENT);
         authenticationData.setAuthnSuccess(false);
-        authenticationData.setRemoteIp(IdentityUtil.getClientIpAddress(request));
+        if (request != null) {
+            authenticationData.setRemoteIp(IdentityUtil.getClientIpAddress(request));
+        } else {
+            authenticationData.setRemoteIp(AuthPublisherConstants.NOT_AVAILABLE);
+        }
```

Comment:
introduce a new method for this to avoid duplication

Reply:
Introduced a new method.